webplugin/js/app/mck-sidebox-1.0.js

```diff
@@ -2480,19 +2480,101 @@ var CURRENT_GROUP_DATA={};
                     if (CURRENT_GROUP_DATA.CHAR_CHECK) {
+                        var warningLength = 199;
+                        var maxLength = 256;
+                        var textBox = $mck_text_box[0]; //using seperate selector for vanilla JS functions
                         if (!(document.getElementById('mck-char-count'))) {
                             warningText.innerHTML += '<span> | </span><span id="mck-char-count"></span>';
                         }
                         var remtxt;
-                        var str = ($mck_text_box.text().toString()).trim();
-                        var len = str.length;
-                        if (len > 199) {
-                            if (len > 256) {
-                                remtxt = "(" + MCK_LABELS['limit.remove'] + " " + (len - 256) + " " + MCK_LABELS['limit.characters'] + ")";
+                        var textVal = mckUtils.textVal(textBox);
+                        var str = textVal.trim();
+                        var textLength = str.length;
+                        if (textLength > warningLength) {
+                            var caretPosition = _this.cursorPosition(textBox);
+                            if (textLength > maxLength) {
+                                _this.disableSendButton();
+                                var selection;
+                                remtxt = "(" + MCK_LABELS['limit.remove'] + " " + (textLength - maxLength) + " " + MCK_LABELS['limit.characters'] + ")";
+                                var preSpanContent = str.slice(0,maxLength);
+                                preSpanContent = preSpanContent.replace(/(?:\r\n|\r|\n)/g, '<br>');
+                                var spanContent = str.slice(maxLength);
+                                spanContent = spanContent.replace(/(?:\r\n|\r|\n)/g, '<br>');
+                                var warningSpan = document.getElementById('mck-text-warning-span');
+                                var normalSpan = document.getElementById('mck-normal-span');
+                                if (warningSpan && !(event.type == 'paste')) {
+                                    if(caretPosition <= (maxLength + 1)) {
+                                        warningSpan.innerHTML = spanContent;
+                                        normalSpan.innerHTML = preSpanContent;
+                                        selection = document.getSelection();
+                                        (caretPosition > maxLength) ? selection.collapse(warningSpan.firstChild,caretPosition - maxLength) : selection.collapse(normalSpan.firstChild,caretPosition);
+                                    }
+                                } else if(warningSpan && (event.type == 'paste')) {
+                                    textBox.innerHTML = '<span id = "mck-normal-span"></span><span contenteditable="true" style="background-color:rgb(255,184,194);" id="mck-text-warning-span"></span>';
```

Comment:
@hardikparashar can this "background-color:rgb(255,184,194);" be moved to a css file?